tests/upgrades/test_client.py

```diff
@@ -27,373 +28,303 @@
-                'OS {} is not provided!'.format(client_os.lower())
-            )
-        repo_name = f'{product_name}_repo'
-        tools_prod = entities.Product(organization=org.id).search(
-            query={'search': f'name={product_name}'}
-        )
-        if not tools_prod:
-            tools_prod = entities.Product(name=product_name, organization=org.id).create()
-            tools_repo = entities.Repository(
-                name=repo_name, product=tools_prod, url=tools_repo_url, content_type='yum'
-            ).create()
-            tools_repo.sync()
-        else:
-            tools_repo = entities.Repository(organization=org.id, product=tools_prod).search(
-                query={'search': f'name={repo_name}'}
-            )
-    tools_cv = entities.ContentView(
-        name=ak_name + '_cv', label=ak_name + '_cv', organization=org.id
+@pytest.fixture(scope='module')
+def pre_upgrade_repo(default_org):
+    """Enable custom errata repository"""
+    pre_upgrade_repo = entities.Repository(
+        url=FAKE_0_YUM_REPO, product=entities.Product(organization=default_org).create()
     ).create()
-    tools_cv.repository = [tools_repo, rhel_repo]
-    tools_cv = tools_cv.update(['repository'])
-    tools_cv.publish()
-    tools_cv = tools_cv.read()  # Published CV with new version
-    # Promote CV
-    cvv = entities.ContentViewVersion(id=max([cvv.id for cvv in tools_cv.version])).read()
-    cvv.promote(data={'environment_id': environment.id, 'force': False})
-    tools_ak = entities.ActivationKey(
-        name=ak_name, content_view=tools_cv, organization=org.id, environment=environment
-    ).create()
-    if sat_state == 'pre':
-        tools_sub = 'Red Hat Satellite Employee Subscription'
-        tools_content = 'rhel-{}-server-satellite-tools-{}-rpms'.format(client_os[-1], from_ver)
-    else:
-        tools_sub = tools_prod.name
-    tools_subscription = entities.Subscription(organization=org.id).search(
-        query={'search': f'name="{tools_sub}"', 'per_page': 1000}
-    )[0]
-    rhel_subscription = entities.Subscription(organization=org.id).search(
-        query={'search': f'name={rhel_prod.name}', 'per_page': 1000}
-    )[0]
-    tools_ak.add_subscriptions(data={'subscription_id': tools_subscription.id})
-    if sat_state == 'pre':
-        tools_ak.content_override(
-            data={'content_override': {'content_label': tools_content, 'value': '1'}}
-        )
-    tools_ak.add_subscriptions(data={'subscription_id': rhel_subscription.id})
-    return tools_ak
+    assert pre_upgrade_repo.sync()['result'] == 'success'
+    return pre_upgrade_repo
 
 
-def create_yum_test_repo(product_name, repo_url, org):
-    """Creates yum repo from given url and syncs
+@pytest.fixture(scope='module')
+def post_upgrade_repo(default_org):
+    """Enable custom errata repository"""
+    post_upgrade_repo = entities.Repository(
+        url=FAKE_9_YUM_REPO, product=entities.Product(organization=default_org).create()
```

Comment:
Same here, please use `module_product` fixture.

Reply:
OK, done that